spec/unit/worker_async_spec.rb

```diff
@@ -43,6 +44,54 @@ module Backend
             expect(test_job).to have_received(:fail).with(error)
           end
         end
+
+        describe 'when there are pending jobs in the memory queue' do
+          subject do
+            WorkerAsync.new
+          end
+
+          let(:test_job) { instance_double('BackgroundJob') }
+          let(:n_reports) { 10 }
+          let(:queue) { Queue.new.tap { |q| n_reports.times.map { q.push test_job } } }
+
+          before do
+            subject.instance_variable_set(:@jobs, queue)
+          end
+
+          it 'clears the queue'do
+            expect(subject).to receive(:perform).exactly(n_reports).times.with(test_job)
+            queue.close
+
+            subject.send(:clear_queue)
+          end
+
+          it 'clears the queue also when new jobs are added during the execution' do
```

Comment:
I think this one deserves a comment along the lines of:
```
  # this is not an expected real usage scenario but we test it anyway
```

Just as a courtesy to future readers (maybe ourselves) that this may not be a hard requirement (subject to additional investigation).